x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Regression.java

```diff
@@ -201,8 +202,8 @@ public boolean supportsCategoricalFields() {
     }
 
     @Override
-    public Set<String> getRequiredFields() {
-        return Collections.singleton(dependentVariable);
+    public List<RequiredField> getRequiredFields() {
+        return Collections.singletonList(new RequiredField(dependentVariable, new TreeSet<>(Types.numerical())));
```

Comment:
Since you are already using `Collections.singletonList` in this line, could you also use `Collections.singleton` instead of `new TreeSet<>`?

Reply:
Actually, I can simply pass `Types.numerical()` directly. I'll do that.